pgjdbc/src/main/java/org/postgresql/copy/CopyManager.java

```diff
@@ -48,26 +48,24 @@ public CopyManager(BaseConnection connection) throws SQLException {
   }
 
   public CopyIn copyIn(String sql) throws SQLException {
-    CopyOperation op = null;
-    try {
-      op = queryExecutor.startCopy(sql, connection.getAutoCommit());
+    CopyOperation op = queryExecutor.startCopy(sql, connection.getAutoCommit());
+    if (op == null || op instanceof CopyIn) {
```

Comment:
I'm still of the opinion that the code path is faster using the exception
as it truly should be an exception.

Dave Cramer

On 5 March 2016 at 07:17, Philippe Marschall <notifications@github.com>
wrote:

> In pgjdbc/src/main/java/org/postgresql/copy/CopyManager.java
> <https://github.com/pgjdbc/pgjdbc/pull/527#discussion_r55120290>:
>
> > @@ -48,26 +48,24 @@ public CopyManager(BaseConnection connection) throws SQLException {
> >    }
> >
> >    public CopyIn copyIn(String sql) throws SQLException {
> > -    CopyOperation op = null;
> > -    try {
> > -      op = queryExecutor.startCopy(sql, connection.getAutoCommit());
> > +    CopyOperation op = queryExecutor.startCopy(sql, connection.getAutoCommit());
> > +    if (op == null || op instanceof CopyIn) {
>
> That was the behaviour of the old code. I wanted to keep this PR as a
> refactoring in the actual sense, not change the behaviour of the code.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pgjdbc/pgjdbc/pull/527/files#r55120290>.
>